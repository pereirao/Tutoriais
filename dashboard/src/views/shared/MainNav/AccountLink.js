// @flow

import React from 'react';
import {connect} from 'react-redux';
import {Dropdown} from 'semantic-ui-react';
import I18n from '../../../localization';
import {session_actions, session_selectors} from '../../../business/session';

import style_sheet from './MainNav.module.css';

const {myName} = session_selectors;
const {signOut} = session_actions;

type AccountLinkProps = { name: string, signMeOut: () => void };
export const AccountLink = ({name, signMeOut}: AccountLinkProps) => (
  <div>
    <Dropdown compact text={name}>
      <Dropdown.Menu className={style_sheet.accountLinkDropdownMenu}>
        <Dropdown.Item text="Sign Out" icon="log out" onClick={signMeOut}/>
      </Dropdown.Menu>
    </Dropdown>
    <p className={style_sheet.accountRole}>{I18n.t('ui.header.employee')}</p>
  </div>
);


const AccountLinkSTP = state => ({name: myName(state)});
const AccountLinkDTP = {signMeOut: signOut};
const AccountLinkContainer = connect(AccountLinkSTP, AccountLinkDTP)(AccountLink);

export default AccountLinkContainer;
