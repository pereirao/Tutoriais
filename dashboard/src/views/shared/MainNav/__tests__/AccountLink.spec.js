import React from 'react';

import { AccountLink } from '../AccountLink';

it('renders AccountLink correctly', () => {
  expect(shallow(<AccountLink name="Tim Jacklepappy" />)).toMatchSnapshot();
});
