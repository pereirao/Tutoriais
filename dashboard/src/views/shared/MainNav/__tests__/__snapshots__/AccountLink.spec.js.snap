// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AccountLink correctly 1`] = `
<div>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    compact={true}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    selectOnBlur={true}
    text="Tim Jacklepappy"
  >
    <DropdownMenu
      className="accountLinkDropdownMenu"
    >
      <DropdownItem
        icon="log out"
        text="Sign Out"
      />
    </DropdownMenu>
  </Dropdown>
  <p
    className="accountRole"
  >
    employee
  </p>
</div>
`;
